app/collection.go

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package app
+
+import (
+	"net/http"
+
+	"github.com/mattermost/mattermost-server/v6/model"
+	"github.com/mattermost/mattermost-server/v6/utils"
+)
+
+func (a *App) registerCollectionAndTopic(pluginID, collectionType, topicType string) error {
+	if !a.Config().FeatureFlags.ThreadsEverywhere {
```

Comment:
I wonder if we should omit the feature flag check here: the intent of a feature flag is to be able to toggle it on and off in real-time, but we have no way of having a plugin re-register collections and topics when that happens, short of a server restart.

I think it's probably fine to support registration even if the feature is disabled. Downstream code will definitely use this feature flag, of course.

Reply:
good thinking